lib/private_address_check/tcpsocket_ext.rb

```diff
@@ -12,12 +12,31 @@ def only_public_connections
 end
 
 TCPSocket.class_eval do
+
+  EXCEPTION_WHITELIST = []
```

Comment:
Can you share what sort of exceptions you would add to this whitelist for your use case? I am wondering if there should be a default set. Also, this may be better to make it a per-thread configuration similar to the existing check flag vs being global state.